bundle/permissions/workspace_root.go

```diff
@@ -52,16 +53,54 @@ func giveAccessForWorkspaceRoot(ctx context.Context, b *bundle.Bundle) error {
 	}
 
 	w := b.WorkspaceClient().Workspace
-	obj, err := w.GetStatusByPath(ctx, b.Config.Workspace.RootPath)
+	err := setPermissions(ctx, w, b.Config.Workspace.RootPath, permissions)
 	if err != nil {
 		return err
 	}
 
-	_, err = w.UpdatePermissions(ctx, workspace.WorkspaceObjectPermissionsRequest{
+	if !strings.HasPrefix(b.Config.Workspace.ArtifactPath, b.Config.Workspace.RootPath) {
```

Comment:
The prefix check needs to use the root path with trailing slash to make sure it is nested under it.

There is an edge case for:
* root path is `/foo`
* artifact path is `/foobar`